### PR TITLE
[PREG-183] Remove async mode from pregel

### PR DESCRIPTION
**Pull request by jvolmer**
Deletes async code - including AsyncSCC and related code. It is not tested and the code is hard to follow with it because it is hacked in.

(Code PR: https://github.com/arangodb/arangodb/pull/17802)

### CODE REVIEW

**Comment by nerpaula** [comment]
@jvolmer this applies for 3.10 as well? or only 3.11?

**Comment by jvolmer** [comment]
@nerpaula Only to 3.11

File: 3.11/release-notes-upgrading-changes311.md [hunk 1]
```diff
@@ -44,6 +44,11 @@ version of ArangoDB. Setting the option to anything but the value of
 From v3.11.0 onwards, this option is deprecated, and setting it to a value
 different than the value of `--agency.size` leads to a startup error.
 
+### Pregel options
+
+The `asyinc` option used in algorithms which support an asynchronous mode
+to run without synchronized global iterations has been removed.
```
Comment (Simran-B) [inline 3.11/release-notes-upgrading-changes311.md:50]:
```suggestion
The `async` option has been removed. Some algorithms supported an asynchronous mode
to run without synchronized global iterations. This is no longer supported.
```

**Review by Simran-B — APPROVED** [review]

**Review by nerpaula — APPROVED** [review]
